### ISSUE TITLE
Elastic Search Templates for elasticsearch_dynamic

### ISSUE THREAD

**Issue by willysamz** [uken/fluent-plugin-elasticsearch#241]
#### Problem
When using template_name and template_file or the templates hash. Fluentd fails to start and throws an error. If i comment out these two template configuration settings fluentd/elasticsearch plugin works as expected

```
  2017-02-13 21:20:16 +0000 [error]: unexpected error error="wrong number of arguments (0 for 1)"
  2017-02-13 21:20:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-elasticsearch-1.8.0/lib/fluent/plugin/out_elasticsearch_dynamic.rb:36:in `client'
  2017-02-13 21:20:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-elasticsearch-1.8.0/lib/fluent/plugin/elasticsearch_index_template.rb:12:in `template_exists?'
  2017-02-13 21:20:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-elasticsearch-1.8.0/lib/fluent/plugin/elasticsearch_index_template.rb:23:in `template_install'
  2017-02-13 21:20:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-elasticsearch-1.8.0/lib/fluent/plugin/out_elasticsearch.rb:89:in `configure'
  2017-02-13 21:20:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-elasticsearch-1.8.0/lib/fluent/plugin/out_elasticsearch_dynamic.rb:21:in `configure'
  2017-02-13 21:20:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluentd-0.12.31/lib/fluent/agent.rb:133:in `add_match'`
...
```

#### Steps to replicate
td-agent-gem install --no-document fluent-plugin-elasticsearch -v 1.9.2
elasticsearch version 2.3.5
fluentd v0.12.31

```aconf
<match **>
   type elasticsearch_dynamic
   log_level info
   include_tag_key true
   template_name logstash
   template_file /etc/td-agent/elastic_tpl.json
   host elasticsearch
   port 9200
   buffer_chunk_limit 2M
   logstash_format true
   logstash_prefix logstash-${(record['level'] == nil) ? 'info' : record['level'].to_s.downcase}
   buffer_queue_limit 32
   flush_interval 5s
   max_retry_wait 30
   disable_retry_limit
   num_threads 8
   time_key @timestamp
</match>
```

**Comment by cosmo0920** [comment]
I could reproduce this issue, but it seems not to be working since #122 is merged.

##### logstash_template.json

```json
{
  "template" : "logstash-*",
  "version" : 60001,
  "settings" : {
    "index.refresh_interval" : "5s"
  },
  "mappings" : {
    "_default_" : {
      "dynamic_templates" : [ {
        "message_field" : {
          "path_match" : "message",
          "match_mapping_type" : "string",
          "mapping" : {
            "type" : "text",
            "norms" : false
          }
        }
      }, {
        "string_fields" : {
          "match" : "*",
          "match_mapping_type" : "string",
          "mapping" : {
            "type" : "text", "norms" : false,
            "fields" : {
              "keyword" : { "type": "keyword", "ignore_above": 256 }
            }
          }
        }
      } ],
      "properties" : {
        "@timestamp": { "type": "date"},
        "@version": { "type": "keyword"},
        "geoip"  : {
          "dynamic": true,
          "properties" : {
            "ip": { "type": "ip" },
            "location" : { "type" : "geo_point" },
            "latitude" : { "type" : "half_float" },
            "longitude" : { "type" : "half_float" }
          }
        }
      }
    }
  }
}
```

##### fluent.conf

```aconf
<source>
  @type forward
</source>

<match es.**>
   @type elasticsearch_dynamic
   @log_level info
   include_tag_key true
   template_name logstash-template
   template_file logstash-template.json
   host localhost
   port 9200
   logstash_format true
   logstash_prefix logstash-${(record['level'] == nil) ? 'info' : record['level'].to_s.downcase}
   <buffer>
     @type memory
     flush_interval 5s
     queue_limit_length 32
     chunk_limit_size 2M
     retry_forever
     flush_thread_count 8
     retry_max_interval 30
   </buffer>
   time_key @timestamp
</match>
```

**Comment by denniswebb** [comment]
I can reproduce still with ES plugin `1.13.1` and FluentD `v0.12.34`

Any known workaround?

**Comment by cosmo0920** [comment]
> Any known workaround?

Nope.

**Comment by cosmo0920** [comment]
For one ES server, I've found a solution to work `@elasticsearch_dynamic`.
Could anyone confirm above patch?

**Comment by cosmo0920** [comment]
I've released above patch as v1.13.2 and v2.6.1.
If anyone hit another issue in above patch, please open an another issue.
